text/0000-build-std/5-stage-1b.md

```diff
@@ -0,0 +1,986 @@
+
+*See the following sections for rationale/alternatives:*
+
+- [*Why permit patching of the standard library dependencies on nightly?*][rationale-patching]
+
+*See the following sections for relevant unresolved questions:*
+
+- [*What syntax is used to patch dependencies on the standard library in `Cargo.toml`?*][unresolved-patch-syntax]
+
+## Features
+[features]: #features
+
+On a stable toolchain, it is not permitted to enable or disable features of
+explicit standard library dependencies ([?][rationale-features]), as in the
+below example:
+
+```toml
+[package]
+name = "hello_world"
+version = "0.1.0"
+edition = "2024"
+
+[dependencies]
+std = { builtin = true, features = [ "foo" ] } # not permitted
+# ..or..
+std = { builtin = true, default-features = false } # not permitted
+```
+
+*See the following sections for rationale/alternatives:*
+
+- [*Why limit enabling standard library features to nightly?*][rationale-features]
+
+*See the following sections for future possibilities:*
+
+- [*Allow enabling/disabling features with build-std*][future-features]
+
+## Public and private dependencies
+[public-and-private-dependencies]: #public-and-private-dependencies
+
+Implicit dependencies on the standard library default to being public
+dependencies ([?][rationale-implicit-public]). When a standard library is
+explicitly written, then it will be private by default, like any other written
+dependency, unless explicitly marked as public ([?][rationale-explicit-private]).
+
+```toml
+[package]
+name = "hello_world"
+version = "0.1.0"
+edition = "2024"
+
+[dependencies]
+```
+
+..is equivalent to the following explicit dependency on `std`:
+
+```toml
+[package]
+name = "hello_world"
+version = "0.1.0"
+edition = "2024"
+
+[dependencies]
+std = { builtin = true, public = true }
+```
+
+*See the following sections for rationale/alternatives:*
+
+- [*Why default to public for the implicit standard library dependencies?*][rationale-implicit-public]
+- [*Why follow the default privacy of explicit standard library dependencies?*][rationale-explicit-private]
+
+## `dev-dependencies` and `build-dependencies`
+[dev-dependencies-and-build-dependencies]: #dev-dependencies-and-build-dependencies
+
+Explicit dependencies on the standard library are not supported in
+`build-dependencies` ([?][rationale-no-deps-in-build-deps]).
+
+Implicit and explicit dependencies on the standard library are supported for
+`dev-dependencies` in the same way as regular `dependencies`.
+
+## Registries
+[registries]: #registries
+
+Standard library dependencies will be present in the registry index
+([?][rationale-cargo-index]). A `builtin_deps` key is added to the
+[index's JSON schema][cargo-json-schema] ([?][rationale-cargo-builtindeps]).
+`builtin_deps` is similar to the existing `deps` key and contains a list of JSON
+objects, each representing a dependency that is "builtin" to the Rust toolchain
+and cannot otherwise be found in the registry.
```

Comment:
The traditional way to populate the Index entry is to translate it from the [publish request](https://doc.rust-lang.org/cargo/reference/registry-web-api.html#publish).  However, we are (slowly) working to encourage people to do otherwise.

Do we also need to extend that API endpoint?

Reply:
I expect we would, will note this

Reply:
Mentioned publish endpoint in 9f8f4ff6d5e4abfe2dc4858e110b7e1d796f9562

Reply:
Thanks!